### PR TITLE
[KYUUBI #1923] Add GetFunctions for trino engine

### PR DESCRIPTION
**Pull request by SteNicholas**
### _Why are the changes needed?_

Add GetFunctions for trino engine.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by SteNicholas** [comment]
@hddong @yaooqinn @pan3793, could you please help to review this pull request?

File: externals/kyuubi-trino-engine/src/main/scala/org/apache/kyuubi/engine/trino/operation/TrinoOperationManager.scala [hunk 1]
```diff
@@ -91,5 +91,8 @@ class TrinoOperationManager extends OperationManager("TrinoOperationManager") {
       session: Session,
       catalogName: String,
       schemaName: String,
-      functionName: String): Operation = null
+      functionName: String): Operation = {
```
Comment (yaooqinn) [inline externals/kyuubi-trino-engine/src/main/scala/org/apache/kyuubi/engine/trino/operation/TrinoOperationManager.scala:94]: just raise `KyuubiSQLException.featureNotSupported` for simplication

**Review by yaooqinn — APPROVED** [review]
LGTM

**Comment by yaooqinn** [comment]
thanks, merged to master

**Pull request closed without merging**